build.sbt

```diff
@@ -26,3 +28,35 @@ lazy val toolkit = crossProject(JVMPlatform, JSPlatform, NativePlatform)
     ),
     mimaPreviousArtifacts := Set()
   )
+
+lazy val docs = project
+  .in(file("site"))
+  .enablePlugins(TypelevelSitePlugin)
+  .dependsOn(
+    toolkit.jvm // overkill? this way we can use mdoc to check the examples at least...
+  )
+  .settings(
+    scalaVersion := "3.2.2",
```

Comment:
This will force to write examples with scala 3 syntax, but we want to encourage this transition right? :D

Reply:
Hmm, it won't force anything without the additional scalac options :) anyway this is unnecessary, scala 3 should already be the default.

Reply:
Well, without it I get a bunch of these
```
error: examples.md:23:22: 
expression expected but '[' found
  given EntityDecoder[IO, Data] = jsonOf[IO, Data]
                     ^
error: examples.md:83:3: 
indented definitions expected, eof found
  def main = (prefix, delimiter, suffix).mapN { (pre, delim, post) =>
  ^^^
```

Reply:
Huh, so it does. That seems like a bug.

```
sbt:root> show docs/scalaVersion
[info] 3.2.2
sbt:root> docs/mdoc/scalaVersion
[info] 3.2.2
sbt:root> docs/mdoc
[info] running mdoc.SbtMain 
info: Compiling 3 files to /workspace/toolkit/site/target/mdoc
error: examples.md:21:3: 
...
info: Compiled in 7.06s (8 errors)
[error] java.lang.RuntimeException: mdoc failed
[error]         at scala.sys.package$.error(package.scala:27)
[error]         at mdoc.SbtMain$.main(Main.scala:30)
[error]         at mdoc.SbtMain.main(Main.scala)
[error]         at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
[error]         at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
[error]         at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
[error]         at java.base/java.lang.reflect.Method.invoke(Method.java:566)
[error] stack trace is suppressed; run last docs / Compile / runMain for the full output
[error] (docs / Compile / runMain) mdoc failed
[error] Total time: 9 s, completed Mar 4, 2023, 5:11:17 PM
```

Reply:
Oh I see, it's because it's overriding `crossScalaVersions` implicitly to just include Scala 3, which then disables the Scala 2 compat mode.

Anyway, if you want to do this we might as well drop 2.12 and 2.13 😛 another option is to have tabs, but that might be a lot of work.
https://typelevel.org/otel4s/examples/jaeger-docker.html#project-setup

Reply:
I'm not sure I'm following. If there's the chance to support every Scala version, I'll pick it, but I'm not sure on how to do that.
Also, tabs are fine and if someone wants to enrich his/her examples I'll be glad to accept the PR :D

Reply:
> If there's the chance to support every Scala version, I'll pick it, but I'm not sure on how to do that.

I must be confused too :) if you want to support every Scala version, using Scala 3-only syntax probably won't work :)

Reply:
I'm even more confused now. What's the purpose of making it compile with every Scala version if I have to rely on Scala 2's syntax? 🤣